### PR TITLE
Adding process page to view and update current process

### PR DESCRIPTION
**Pull request by gautamguptabasant**
Screenshot 2020-12-13 at 9 09 30 PM
Screenshot 2020-12-13 at 9 10 16 PM
Screenshot 2020-12-13 at 9 10 25 PM
Screenshot 2020-12-13 at 9 10 33 PM

### CODE REVIEW

File: src/Pages/FieldInfoPage.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import {
+  Button,
```
Comment (mahimavuppuluri) [inline src/Pages/FieldInfoPage.js]: remove this button

Comment (mahimavuppuluri) [reply src/Pages/FieldInfoPage.js]: "Fetch" button not needed anymore, if its being used in code, remove it. location is filtered on selecting something from drop down

File: src/Pages/ProcessPage.js [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+const ProcessPage = () => {
+  const classes = useStyles();
+  const [locationCode, setLocationCode] = useState('');
+  const [locations, setLocations] = useState([]);
+  const [fieldId, setFieldId] = useState('');
+  const [fields, setFields] = useState([]);
+  const [cropCycleId, setCropCycleId] = useState('');
+  const [cropCycles, setCropCycles] = useState([]);
+  const [processStatuses, setPprocessStatuses] = useState([]);
```
Comment (mahimavuppuluri) [inline src/Pages/ProcessPage.js]: minor: typo in the setter

File: src/Pages/ProcessPage.js [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+  const [alertMessage, setAlertMessage] = useState('');
+  const [alertStatus, setAlertStatus] = useState(false);
+  const [alertSeverity, setAlertSeverity] = useState('');
+
+  const handleAlertClose = (_event, reason) => {
```
Comment (mahimavuppuluri) [inline src/Pages/ProcessPage.js:63]: functions related to alert are there in every page, should we make it a separate component?

**Review by mahimavuppuluri — APPROVED** [review]

**Comment by aashnaPrakash** [comment]
end date can be lesser than start date.
image
Are we planning to take this up as form-validation?

**Comment by aashnaPrakash** [comment]
I select locality - select field - select crop cycle - change the status to in progress and select start and end dates.
Now if I just change the crop cycle, should the card not update?

**Comment by aashnaPrakash** [comment]
index.js:1 Warning: Each child in a list should have a unique "key" prop.

Check the render method of `ProcessPage`. See https://reactjs.org/link/warning-keys for more information.
    in ProcessCard (at ProcessPage.js:215)
    in ProcessPage (at ApplnBar.js:122)
    in div (created by Styled(MuiBox))
    in Styled(MuiBox) (at DynamicTabs.js:17)
    in div (at DynamicTabs.js:10)
    in TabPanel (at DynamicTabs.js:67)
    in div (at DynamicTabs.js:53)
    in DynamicTabs (at ApplnBar.js:99)
    in div (at ApplnBar.js:129)
    in ApplnBar (at App.js:9)
    in div (at App.js:8)
    in App (at src/index.js:9)

**Comment by aashnaPrakash** [comment]
Warning: React does not recognize the `KeyboardButtonProps` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `keyboardbuttonprops` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in div (created by ForwardRef(FormControl))
    in ForwardRef(FormControl) (created by WithStyles(ForwardRef(FormControl)))
    in WithStyles(ForwardRef(FormControl)) (created by ForwardRef(TextField))
    in ForwardRef(TextField) (created by WithStyles(ForwardRef(TextField)))
    in WithStyles(ForwardRef(TextField)) (created by PureDateInput)
    in PureDateInput (created by ModalWrapper)
    in ModalWrapper (created by Wrapper)
    in Wrapper (created by PickerWithState)
    in PickerWithState (at ProcessCard.js:196)
    in div (created by ForwardRef(Grid))
    in ForwardRef(Grid) (created by WithStyles(ForwardRef(Grid)))
    in WithStyles(ForwardRef(Grid)) (at ProcessCard.js:195)
    in div (created by ForwardRef(Grid))
    in ForwardRef(Grid) (created by WithStyles(ForwardRef(Grid)))
    in WithStyles(ForwardRef(Grid)) (at ProcessCard.js:160)
    in MuiPickersUtilsProvider (at ProcessCard.js:159)
    in div (created by ForwardRef(AccordionDetails))
    in ForwardRef(AccordionDetails) (created by WithStyles(ForwardRef(AccordionDetails)))
    in WithStyles(ForwardRef(AccordionDetails)) (at ProcessCard.js:158)
    in div (created by ForwardRef(Accordion))
    in div (created by Transition)
    in div (created by Transition)
    in div (created by Transition)
    in Transition (created by ForwardRef(Collapse))
    in ForwardRef(Collapse) (created by WithStyles(ForwardRef(Collapse)))
    in WithStyles(ForwardRef(Collapse)) (created by ForwardRef(Accordion))
    in div (created by ForwardRef(Paper))
    in ForwardRef(Paper) (created by WithStyles(ForwardRef(Paper)))
    in WithStyles(ForwardRef(Paper)) (created by ForwardRef(Accordion))
    in ForwardRef(Accordion) (created by WithStyles(ForwardRef(Accordion)))
    in WithStyles(ForwardRef(Accordion)) (at ProcessCard.js:120)
    in div (at ProcessCard.js:119)
    in ProcessCard (at ProcessPage.js:215)
    in div (created by ForwardRef(Grid))
    in ForwardRef(Grid) (created by WithStyles(ForwardRef(Grid)))
    in WithStyles(ForwardRef(Grid)) (at ProcessPage.js:305)
    in div (created by ForwardRef(Grid))
    in ForwardRef(Grid) (created by WithStyles(ForwardRef(Grid)))
    in WithStyles(ForwardRef(Grid)) (at ProcessPage.js:229)
    in div (created by ForwardRef(FormGroup))
    in ForwardRef(FormGroup) (created by WithStyles(ForwardRef(FormGroup)))
    in WithStyles(ForwardRef(FormGroup)) (at ProcessPage.js:228)
    in ProcessPage (at ApplnBar.js:122)
    in div (created by Styled(MuiBox))
    in Styled(MuiBox) (at DynamicTabs.js:17)
    in div (at DynamicTabs.js:10)
    in TabPanel (at DynamicTabs.js:67)
    in div (at DynamicTabs.js:53)
    in DynamicTabs (at ApplnBar.js:99)
    in div (at ApplnBar.js:129)
    in ApplnBar (at App.js:9)
    in div (at App.js:8)
    in App (at src/index.js:9)
console.<computed> @ index.js:1

**Comment by aashnaPrakash** [comment]
In case I switch to a locality which does not have fields, it always shows the crop cycles from the last selected field.

File: src/Pages/FieldInfoPage.js (hunk 1 above)
Comment (gautamguptabasant) [reply src/Pages/FieldInfoPage.js]: Done

File: src/Pages/ProcessPage.js (hunk 2 above)
Comment (gautamguptabasant) [reply src/Pages/ProcessPage.js]: Updated, thanks

**Comment by gautamguptabasant** [comment]
> end date can be lesser than start date.
> image
> Are we planning to take this up as form-validation?

As discussed, we will enable form validations separately. It's added in the API side as of now.

**Comment by gautamguptabasant** [comment]
> I select locality - select field - select crop cycle - change the status to in progress and select start and end dates.
> Now if I just change the crop cycle, should the card not update?

Good catch. Updated the impl to handle this case.

**Comment by gautamguptabasant** [comment]
> index.js:1 Warning: Each child in a list should have a unique "key" prop.
> 
> Check the render method of `ProcessPage`. See https://reactjs.org/link/warning-keys for more information.
> in ProcessCard (at ProcessPage.js:215)
> in ProcessPage (at ApplnBar.js:122)
> in div (created by Styled(MuiBox))
> in Styled(MuiBox) (at DynamicTabs.js:17)
> in div (at DynamicTabs.js:10)
> in TabPanel (at DynamicTabs.js:67)
> in div (at DynamicTabs.js:53)
> in DynamicTabs (at ApplnBar.js:99)
> in div (at ApplnBar.js:129)
> in ApplnBar (at App.js:9)
> in div (at App.js:8)
> in App (at src/index.js:9)

Resolved

**Comment by gautamguptabasant** [comment]
> Warning: React does not recognize the `KeyboardButtonProps` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `keyboardbuttonprops` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
> in div (created by ForwardRef(FormControl))
> in ForwardRef(FormControl) (created by WithStyles(ForwardRef(FormControl)))
> in WithStyles(ForwardRef(FormControl)) (created by ForwardRef(TextField))
> in ForwardRef(TextField) (created by WithStyles(ForwardRef(TextField)))
> in WithStyles(ForwardRef(TextField)) (created by PureDateInput)
> in PureDateInput (created by ModalWrapper)
> in ModalWrapper (created by Wrapper)
> in Wrapper (created by PickerWithState)
> in PickerWithState (at ProcessCard.js:196)
> in div (created by ForwardRef(Grid))
> in ForwardRef(Grid) (created by WithStyles(ForwardRef(Grid)))
> in WithStyles(ForwardRef(Grid)) (at ProcessCard.js:195)
> in div (created by ForwardRef(Grid))
> in ForwardRef(Grid) (created by WithStyles(ForwardRef(Grid)))
> in WithStyles(ForwardRef(Grid)) (at ProcessCard.js:160)
> in MuiPickersUtilsProvider (at ProcessCard.js:159)
> in div (created by ForwardRef(AccordionDetails))
> in ForwardRef(AccordionDetails) (created by WithStyles(ForwardRef(AccordionDetails)))
> in WithStyles(ForwardRef(AccordionDetails)) (at ProcessCard.js:158)
> in div (created by ForwardRef(Accordion))
> in div (created by Transition)
> in div (created by Transition)
> in div (created by Transition)
> in Transition (created by ForwardRef(Collapse))
> in ForwardRef(Collapse) (created by WithStyles(ForwardRef(Collapse)))
> in WithStyles(ForwardRef(Collapse)) (created by ForwardRef(Accordion))
> in div (created by ForwardRef(Paper))
> in ForwardRef(Paper) (created by WithStyles(ForwardRef(Paper)))
> in WithStyles(ForwardRef(Paper)) (created by ForwardRef(Accordion))
> in ForwardRef(Accordion) (created by WithStyles(ForwardRef(Accordion)))
> in WithStyles(ForwardRef(Accordion)) (at ProcessCard.js:120)
> in div (at ProcessCard.js:119)
> in ProcessCard (at ProcessPage.js:215)
> in div (created by ForwardRef(Grid))
> in ForwardRef(Grid) (created by WithStyles(ForwardRef(Grid)))
> in WithStyles(ForwardRef(Grid)) (at ProcessPage.js:305)
> in div (created by ForwardRef(Grid))
> in ForwardRef(Grid) (created by WithStyles(ForwardRef(Grid)))
> in WithStyles(ForwardRef(Grid)) (at ProcessPage.js:229)
> in div (created by ForwardRef(FormGroup))
> in ForwardRef(FormGroup) (created by WithStyles(ForwardRef(FormGroup)))
> in WithStyles(ForwardRef(FormGroup)) (at ProcessPage.js:228)
> in ProcessPage (at ApplnBar.js:122)
> in div (created by Styled(MuiBox))
> in Styled(MuiBox) (at DynamicTabs.js:17)
> in div (at DynamicTabs.js:10)
> in TabPanel (at DynamicTabs.js:67)
> in div (at DynamicTabs.js:53)
> in DynamicTabs (at ApplnBar.js:99)
> in div (at ApplnBar.js:129)
> in ApplnBar (at App.js:9)
> in div (at App.js:8)
> in App (at src/index.js:9)
> console. @ index.js:1

Resolved

**Review by aashnaPrakash — APPROVED** [review]

**Pull request merged**